app/actions/nftGalleryActions.js

```diff
@@ -39,39 +31,44 @@ const DEFAULT_NFT_GALLERY_RESULTS = { results: [], page: 0, count: 0 }
 
 export async function parseGhostMarketResults({
   address,
-  page = 0,
+  page = 1,
   previousResults = [],
 }: {
   address: string,
   page: number,
   previousResults: NftGalleryItem[],
 }): Promise<NftGalleryResults> {
   try {
-    const LIMIT = 8
-    const OFFSET = LIMIT * page
+    const SIZE = 8
 
     const response = await axios.get(
-      `https://api.ghostmarket.io/api/v1/assets?chain=n3&owner=${address}&limit=${LIMIT}&offset=${OFFSET}&with_total=1`,
+      `https://api.ghostmarket.io/api/v2/assets?chain=n3&owners[]=${address}&size=${SIZE}&page=${page}&getTotal=true`,
     )
 
-    const items = response?.data?.assets ?? []
-    const count = response?.data?.total_results ?? 0
-    if (items.length) {
-      const results = items.map(asset => {
-        const parsed = {
-          metadata: asset.nft.nft_metadata,
-          series: asset.nft.series,
-          tokenId: asset.nft.token_id,
-          contract: asset.nft.contract,
-          collection: asset.nft.collection,
-        }
-        return parsed
-      })
+    const { assets, total: count } = response?.data
 
-      return { results: previousResults.concat(results), page, count }
-    }
+    if (!assets || !assets.length || !count) throw new Error('No assets found')
```

Comment:
I think keeping the previous pattern of returning an empty payload makes more sense than returning an error here as there is no error that occurred - no NFT's is not really an error

Reply:
Ok. Solved